tachyon/math/base/groups.h

```diff
@@ -132,12 +146,13 @@ class MultiplicativeGroup : public MultiplicativeSemigroup<G> {
 
     // Invert |product|.
     // (a₁ * a₂ * ... *  aₙ)⁻¹
-    G product_inv = product.Inverse();
+    std::optional<G> product_inv = product.Inverse();
+    if (UNLIKELY(!product_inv)) return false;
```

Comment:
Could you add logging here?

```suggestion
    if (UNLIKELY(!product_inv)) {
      LOG(ERROR) << "Inverse of zero attempted";
      return false;
    }
```

Reply:
I believe that the `.Inverse()` above will trickle down to `_inv()`, `.InverseInPlace()`, and `MontgomeryInverse()`, which should have the logging of `LOG(ERROR) << "Inverse of zero attempted";`